src/features/HomePage/index.tsx

```diff
@@ -11,13 +12,21 @@ import FindMentor from './components/FindMentor';
 import Info from './components/Info';
 import NewestMentors from './components/NewestMentors';
 import NewMessages from './components/NewMessages';
+
 import { OUTER_HORIZONTAL_MARGIN } from '@/components/constants';
+
+import ProfileWidget from './components/ProfileWidget';
+
 import PageWithTransition from '@/components/PageWithTransition';
 import Welcome from './components/Welcome';
+import { selectMentorById } from '../MentorPage/mentorPageApi';
 
 const HomePage = () => {
   const hasUnreadMessages = useAppSelector(selectHasUnreadMessages);
   const { isTablet } = useGetLayoutMode();
+  const userRole = useAppSelector(selectUserRole);
+  const isMentor = userRole === 'mentor';
+  const mentor = useAppSelector(selectMentorById(useAppSelector(selectUserId)));
```

Comment:
I would simplify the selector `selectMentorById` by combining the two selectors in following way:

```typescript
export const selectMentorById = createSelector(
  [selectMentors, selectUserId],
  (mentors, buddyId) => (buddyId ? mentors.data?.[buddyId] : undefined),
);
```

Then it can be used

```typescript
  const mentor = useAppSelector(selectMentorById)
```

Reply:
Also the name of the selector... maybe it would be better if it was `selectMyMentorUser` etc? To indicate we are trying to get the current users mentor-profile. Maybe `selectMyMentorProfile`